NTupleAnalyzerHH_nanoAOD.py

```diff
@@ -940,41 +1020,33 @@ def LeptonsAndJetsFromHH(T):
 		else:
 			matchedrecomuons.append(emptyvector)
 			recomuoninds.append(-99)
-		#print 'dr', mindr, 'pt', muons[bestrecomuonind].Pt(), g.Pt(), 'eta', muons[bestrecomuonind].Eta(), g.Eta(), 'phi', muons[bestrecomuonind].Phi(), g.Phi()
 	while len(recomuoninds)<2 :
 		recomuoninds.append(-99)
-	#return([genmuons,matchedrecomuons,recomuoninds])
-	#return(recomuoninds)
 
+###################### genMatching Electrons ##########################
 	for n in range(T.nElectron):
 		m = TLorentzVector()
-		m.SetPtEtaPhiM(T.Electron_pt[n],T.Electron_eta[n],T.Electron_phi[n],0)
+		m.SetPtEtaPhiM(T.Electron_pt[n],T.Electron_eta[n],T.Electron_phi[n],T.Electron_mass[n])
 		electrons.append(m)
-	for n in range(T.nGenPart):
-		pdg = T.GenPart_pdgId[n]
-		if pdg not in [11,-11]:
-			continue
-		motherIndex = T.GenPart_genPartIdxMother[n]
-
-		motherid = 0
-		if motherIndex>-1:
-			motherid = T.GenPart_pdgId[motherIndex]
-			#print 'pdgId:',pdg,'index:',T.GenParticleMotherIndex[n],'mother pdgId:',T.GenParticlePdgId[motherIndex]
-		if motherid not in [23,-23,25,-25]:
-			continue
-		if motherid in [23,-23]: onShellZEle=True
-		#elif v=='': print 'Electron not from Z???  Eleparent:',motherid
-		m = TLorentzVector()
-		m.SetPtEtaPhiM(T.GenPart_pt[n],T.GenPart_eta[n],T.GenPart_phi[n],0.0)
-		genelectrons.append(m)
 
-	#if v=='':print 'onShellZMu:',onShellZMu,'onShellZEle:',onShellZEle
-	#if v == '':print 'number of genmuons:',len(genmuons),'number of genelectrons:',len(genelectrons)
+	if isdata==False:
+
+		for n in range(T.nGenPart):
+			pdg = T.GenPart_pdgId[n]
+			if pdg not in [11,-11]:
+				continue
+			motherIndex = T.GenPart_genPartIdxMother[n]
+
+			motherid = 0
+			if motherIndex>-1:
+				motherid = T.GenPart_pdgId[motherIndex]
+			if motherid not in [23,-23,25,-25]:
+				continue
+			if motherid in [23,-23]: onShellZEle=True
+			m = TLorentzVector()
+			m.SetPtEtaPhiM(T.GenPart_pt[n],T.GenPart_eta[n],T.GenPart_phi[n],T.GenPart_mass[n])
```

Comment:
This looks like a problem in the indentation compared to the old version, please check.